analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -184,4 +179,23 @@ private ProgramState ProcessMethod(SymbolicContext context, IMethodSymbol method
         }
         return state;
     }
+
+    private static CollectionConstraint CollectionCreationConstraint(ProgramState state, IObjectCreationOperationWrapper objectCreation) =>
+        objectCreation.Arguments.SingleOrDefault(x => x.ToArgument().Parameter.Type.DerivesOrImplements(KnownType.System_Collections_IEnumerable)) is { } collectionArgument
+            ? state[collectionArgument]?.Constraint<CollectionConstraint>()
+            : CollectionConstraint.Empty;
+
+    private static NumberConstraint PropertyReferenceConstraint(ProgramState state, IPropertyReferenceOperationWrapper propertyReference)
+    {
+        if (propertyReference.Property.Name is nameof(Array.Length) or nameof(List<int>.Count)
+            && propertyReference.Instance.TrackedSymbol() is { } symbol
```

Comment:
We could use `state.ResolveCapture(propertyReference.Instance).TrackedSymbol()` to cover more cases.